### ISSUE TITLE
[Bug]: Packet error when reading packet: DELETE_ITEM (Only in multiplayer)

### ISSUE THREAD

**Issue by Gogo08190** [mezz/JustEnoughItems#3225]
### Steps to Reproduce the Bug

1. Go on a server
2. Active Cheat mode
3. Take an item from JEI
4. Drag the item back to JEI to delete it

### Expected Behavior

The item is not deleted and causes an error in the server console : 

[13:48:48] [Server thread/ERROR] [me.je.co.ne.ServerPacketRouter/]: Packet error when reading packet: DELETE_ITEM
java.lang.NoClassDefFoundError: Could not initialize class mezz.jei.common.platform.Services
        at mezz.jei.common.network.packets.PacketDeletePlayerItem.readPacketData(PacketDeletePlayerItem.java:50) ~[jei-1.18.2-forge-10.2.1.1004.jar%23197!/:10.2.1.1004] {re:classloading}
        at mezz.jei.common.network.ServerPacketRouter.lambda$onPacket$1(ServerPacketRouter.java:42) ~[jei-1.18.2-forge-10.2.1.1004.jar%23197!/:10.2.1.1004] {re:classloading}
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?] {re:mixin}
        at mezz.jei.common.network.ServerPacketRouter.onPacket(ServerPacketRouter.java:37) ~[jei-1.18.2-forge-10.2.1.1004.jar%23197!/:10.2.1.1004] {re:classloading}
        at mezz.jei.forge.network.NetworkHandler.lambda$registerServerPacketHandler$1(NetworkHandler.java:48) ~[jei-1.18.2-forge-10.2.1.1004.jar%23197!/:10.2.1.1004] {re:classloading,pl:runtimedistcleaner:A}
        at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?] {}
        at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?] {}
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?] {}
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?] {}
        at net.minecraftforge.network.NetworkInstance.dispatch(NetworkInstance.java:68) ~[forge-1.18.2-40.2.1-universal.jar%23377!/:?] {re:classloading}
        at net.minecraftforge.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:78) ~[forge-1.18.2-40.2.1-universal.jar%23377!/:?] {re:classloading}
        at java.util.Optional.map(Optional.java:260) ~[?:?] {re:mixin}
        at net.minecraftforge.network.NetworkHooks.onCustomPayload(NetworkHooks.java:78) ~[forge-1.18.2-40.2.1-universal.jar%23377!/:?] {re:classloading}
        at net.minecraft.server.network.ServerGamePacketListenerImpl.m_7423_(ServerGamePacketListenerImpl.java:1439) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
        at net.minecraft.network.protocol.game.ServerboundCustomPayloadPacket.m_5797_(ServerboundCustomPayloadPacket.java:35) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,re:classloading,pl:mixin:A}
        at net.minecraft.network.protocol.game.ServerboundCustomPayloadPacket.m_5797_(ServerboundCustomPayloadPacket.java:8) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,re:classloading,pl:mixin:A}
        at net.minecraft.network.protocol.PacketUtils.m_131356_(PacketUtils.java:22) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:classloading}
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:classloading,re:mixin}
        at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,re:computing_frames,re:classloading}
        at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:799) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:164) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:782) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:776) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:116) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:761) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:689) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220404.173914-srg.jar%23372!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
        at java.lang.Thread.run(Thread.java:833) [?:?] {re:mixin}
Caused by: java.lang.ExceptionInInitializerError: Exception java.util.ServiceConfigurationError: mezz.jei.common.platform.IPlatformHelper: Provider mezz.jei.forge.platform.PlatformHelper could not be instantiated [in thread "Server thread"]
        at java.util.ServiceLoader.fail(ServiceLoader.java:586) ~[?:?] {}
        at java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:813) ~[?:?] {}
        at java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:729) ~[?:?] {}
        at java.util.ServiceLoader$3.next(ServiceLoader.java:1403) ~[?:?] {}
        at java.util.ServiceLoader.findFirst(ServiceLoader.java:1812) ~[?:?] {}
        at mezz.jei.common.platform.Services.load(Services.java:15) ~[jei-1.18.2-forge-10.2.1.1004.jar%23197!/:10.2.1.1004] {re:classloading}
        at mezz.jei.common.platform.Services.<clinit>(Services.java:11) ~[jei-1.18.2-forge-10.2.1.1004.jar%23197!/:10.2.1.1004] {re:classloading}
        ... 30 more

### Actual Behavior

The item is not deleted and causes an error in the server console

### Extra Notes (Optional)

Minecraft version : 1.18.2
Forge version : Forge - 40.2.4
JEI Version : jei-1.18.2-forge-10.2.1.1004

### latest.log

https://gist.github.com/Gogo08190/e840b41d854d0222940ff3537750edf5

**Comment by SpindlesDev** [comment]
Also having this issue on forge 40.2.0 on latest JEI release (10.2.1.1005 at time of writing)

Console error: 
```
[28Aug2023 01:52:15.354] [Server thread/ERROR] [mezz.jei.common.network.ServerPacketRouter/]: Packet error when reading packet: DELETE_ITEM
java.lang.NoClassDefFoundError: Could not initialize class mezz.jei.common.platform.Services
	at mezz.jei.common.network.packets.PacketDeletePlayerItem.readPacketData(PacketDeletePlayerItem.java:50) ~[jei-1.18.2-forge-10.2.1.1005.jar%2374!/:10.2.1.1005]
	at mezz.jei.common.network.ServerPacketRouter.lambda$onPacket$1(ServerPacketRouter.java:42) ~[jei-1.18.2-forge-10.2.1.1005.jar%2374!/:10.2.1.1005]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at mezz.jei.common.network.ServerPacketRouter.onPacket(ServerPacketRouter.java:37) ~[jei-1.18.2-forge-10.2.1.1005.jar%2374!/:10.2.1.1005]
	at mezz.jei.forge.network.NetworkHandler.lambda$registerServerPacketHandler$1(NetworkHandler.java:48) ~[jei-1.18.2-forge-10.2.1.1005.jar%2374!/:10.2.1.1005]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.network.NetworkInstance.dispatch(NetworkInstance.java:68) ~[forge-1.18.2-40.2.0-universal.jar%23102!/:?]
	at net.minecraftforge.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:78) ~[forge-1.18.2-40.2.0-universal.jar%23102!/:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at net.minecraftforge.network.NetworkHooks.onCustomPayload(NetworkHooks.java:78) ~[forge-1.18.2-40.2.0-universal.jar%23102!/:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.m_7423_(ServerGamePacketListenerImpl.java:1439) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.network.protocol.game.ServerboundCustomPayloadPacket.m_5797_(ServerboundCustomPayloadPacket.java:35) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.network.protocol.game.ServerboundCustomPayloadPacket.m_5797_(ServerboundCustomPayloadPacket.java:8) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.network.protocol.PacketUtils.m_131356_(PacketUtils.java:22) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:799) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:164) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:782) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:776) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:140) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:762) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:689) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220404.173914-srg.jar%2397!/:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Comment by mezz** [comment]
Thanks for the report!
This problem was fixed on 1.19.2 and later https://github.com/mezz/JustEnoughItems/issues/2978, but not on 1.18.2.
I'll backport the fix

**Comment by mezz** [comment]
This is fixed in JEI for 1.18.2 version 10.2.1.1006
